subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -101,6 +103,108 @@ void lll_scan_aux_prepare(void *param)
 	LL_ASSERT(!err || err == -EINPROGRESS);
 }
 
+uint8_t lll_scan_aux_setup(struct lll_scan *lll, struct pdu_adv *pdu,
+			   uint8_t pdu_phy)
+{
+	struct pdu_adv_com_ext_adv *pri_com_hdr;
+	struct pdu_adv_ext_hdr *pri_hdr;
+	struct pdu_adv_aux_ptr *aux_ptr;
+	struct node_rx_pdu *node_rx;
+	uint32_t window_widening_us;
+	uint32_t window_size_us;
+	struct node_rx_ftr *ftr;
+	uint32_t aux_offset_us;
+	uint32_t overhead_us;
+	uint8_t *pri_dptr;
+	uint8_t phy;
+
+	LL_ASSERT(pdu->type == PDU_ADV_TYPE_EXT_IND);
+
+	/* Get reference to extended header */
+	pri_com_hdr = (void *)&pdu->adv_ext_ind;
+	pri_hdr = (void *)pri_com_hdr->ext_hdr_adv_data;
+	pri_dptr = pri_hdr->data;
+
+	/* traverse through adv_addr, if present */
+	if (pri_hdr->adv_addr) {
+		pri_dptr += BDADDR_SIZE;
+	}
+
+	/* traverse through tgt_addr, if present */
+	if (pri_hdr->tgt_addr) {
+		pri_dptr += BDADDR_SIZE;
+	}
+
+	/* traverse through cte_info, if present */
+	if (pri_hdr->cte_info) {
+		pri_dptr += sizeof(struct pdu_cte_info);
+	}
+
+	/* traverse through adi, if present */
+	if (pri_hdr->adi) {
+		pri_dptr += sizeof(struct pdu_adv_adi);
+	}
+
+	/* No need to scan further if no aux_ptr filled */
+	aux_ptr = (void *)pri_dptr;
+	if (unlikely(!pri_hdr->aux_ptr || !aux_ptr->offs)) {
+		return 0;
+	}
+
+	/* Determine the window size */
+	if (aux_ptr->offs_units) {
+		window_size_us = OFFS_UNIT_300_US;
+	} else {
+		window_size_us = OFFS_UNIT_30_US;
+	}
+
+	/* Calculate the aux offset from start of the scan window */
+	aux_offset_us = (uint32_t)aux_ptr->offs * window_size_us;
+
+	/* Calculate the window widening that needs to be deducted */
+	if (aux_ptr->ca) {
+		window_widening_us = SCA_DRIFT_50_PPM_US(aux_offset_us);
+	} else {
+		window_widening_us = SCA_DRIFT_500_PPM_US(aux_offset_us);
+	}
+
+	phy = BIT(aux_ptr->phy);
+
+	overhead_us = PKT_AC_US(pdu->len, pdu_phy);
+	overhead_us += lll_radio_rx_ready_delay_get(phy, 1);
+	overhead_us += window_widening_us;
+	overhead_us += EVENT_JITTER_US;
+	overhead_us += EVENT_OVERHEAD_START_US;
+
+	/* TODO: active_to_start feature port */
+	overhead_us += EVENT_OVERHEAD_XTAL_US;
```

Comment:
This is a workaround for https://github.com/zephyrproject-rtos/zephyr/issues/30245

I will remove this when the issue is resolved.